language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -750,4 +767,194 @@ Optional<T>> getDiagPropertyFilterFunction(int propertyIndex) {
         };
         return filterFunction;
     }
+
+    /**
+     * Returns a Code Action for commands.
+     *
+     * @param commandTitle   title of the code action
+     * @param command        command
+     * @param codeActionKind kind of the code action
+     * @return {@link CodeAction}
+     */
+    public static CodeAction createCodeAction(String commandTitle, Command command, String codeActionKind) {
+        List<Diagnostic> diagnostics = new ArrayList<>();
+        CodeAction action = new CodeAction(commandTitle);
+        action.setKind(codeActionKind);
+        action.setCommand(command);
+        action.setDiagnostics(toDiagnostics(diagnostics));
+        return action;
+    }
+
+    /**
+     * Returns a Code action.
+     *
+     * @param commandTitle title of the code action
+     * @param edits        edits to be added in the code action
+     * @param uri          uri
+     * @return {@link CodeAction}
+     */
+    public static CodeAction createCodeAction(String commandTitle, List<TextEdit> edits, String uri) {
+        List<Diagnostic> diagnostics = new ArrayList<>();
+        CodeAction action = new CodeAction(commandTitle);
+        action.setEdit(new WorkspaceEdit(Collections.singletonList(Either.forLeft(
+                new TextDocumentEdit(new VersionedTextDocumentIdentifier(uri, null), edits)))));
+        action.setDiagnostics(toDiagnostics(diagnostics));
+        return action;
+    }
+
+    /**
+     * Returns a Code action.
+     *
+     * @param commandTitle   title of the code action
+     * @param edits          edits to be added in the code action
+     * @param uri            uri
+     * @param codeActionKind kind of the code action
+     * @return {@link CodeAction}
+     */
+    public static CodeAction createCodeAction(String commandTitle, List<TextEdit> edits, String uri,
+                                              String codeActionKind) {
+        CodeAction action = createCodeAction(commandTitle, edits, uri);
+        action.setKind(codeActionKind);
+        return action;
+    }
+
+    /**
+     * Returns a list of text edits based on diagnostics.
+     *
+     * @param positionDetails {@link DiagBasedPositionDetails}
+     * @param context         code action context
+     * @return list of Text Edits
+     */
+    public static List<TextEdit> getDiagBasedTextEditsForImplementMethod(DiagBasedPositionDetails positionDetails,
```

Comment:
We agreed to add an abstract class for the implement method and implement all code actions and move this logic there. Let's do so.

Reply:
Shoot! already added the abstract class. the function here is a duplicate, will remove it.